site/tests/requests/test_record_upgrade_request.py

```diff
@@ -0,0 +1,584 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Zenodo is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Test Zenodo legacy record upgrade request."""
```

Comment:
Fantastic work! Would I would add in the tests is to test that:
- when the request is opened, or declined, the upload can access the files, the community owner cannot.
- when the request is accepted, both can read files.